sympy/core/relational.py

```diff
@@ -1251,190 +1189,6 @@ def is_neq(lhs, rhs):
-        return False  # only Booleans can equal Booleans
-
-    if lhs.is_infinite or rhs.is_infinite:
-        if fuzzy_xor([lhs.is_infinite, rhs.is_infinite]):
-            return False
-        if fuzzy_xor([lhs.is_extended_real, rhs.is_extended_real]):
-            return False
-        if fuzzy_and([lhs.is_extended_real, rhs.is_extended_real]):
-            return fuzzy_xor([lhs.is_extended_positive, fuzzy_not(rhs.is_extended_positive)])
-
-        # Try to split real/imaginary parts and equate them
-        I = S.ImaginaryUnit
-
-        def split_real_imag(expr):
-            real_imag = lambda t: (
-                'real' if t.is_extended_real else
-                'imag' if (I * t).is_extended_real else None)
-            return sift(Add.make_args(expr), real_imag)
-
-        lhs_ri = split_real_imag(lhs)
-        if not lhs_ri[None]:
-            rhs_ri = split_real_imag(rhs)
-            if not rhs_ri[None]:
-                eq_real = Eq(Add(*lhs_ri['real']), Add(*rhs_ri['real']))
-                eq_imag = Eq(I * Add(*lhs_ri['imag']), I * Add(*rhs_ri['imag']))
-                return fuzzy_and(map(fuzzy_bool, [eq_real, eq_imag]))
-
-        # Compare e.g. zoo with 1+I*oo by comparing args
-        arglhs = arg(lhs)
-        argrhs = arg(rhs)
-        # Guard against Eq(nan, nan) -> Falsesymp
-        if not (arglhs == S.NaN and argrhs == S.NaN):
-            return fuzzy_bool(Eq(arglhs, argrhs))
-
-    if all(isinstance(i, Expr) for i in (lhs, rhs)):
-        # see if the difference evaluates
-        dif = lhs - rhs
-        z = dif.is_zero
-        if z is not None:
-            if z is False and dif.is_commutative:  # issue 10728
-                return False
-            if z:
-                return True
-
-        n2 = _n2(lhs, rhs)
-        if n2 is not None:
-            return _sympify(n2 == 0)
-
-        # see if the ratio evaluates
-        n, d = dif.as_numer_denom()
-        rv = None
-        if n.is_zero:
-            rv = d.is_nonzero
-        elif n.is_finite:
-            if d.is_infinite:
-                rv = True
-            elif n.is_zero is False:
-                rv = d.is_infinite
-                if rv is None:
-                    # if the condition that makes the denominator
-                    # infinite does not make the original expression
-                    # True then False can be returned
-                    l, r = clear_coefficients(d, S.Infinity)
-                    args = [_.subs(l, r) for _ in (lhs, rhs)]
-                    if args != [lhs, rhs]:
-                        rv = fuzzy_bool(Eq(*args))
-                        if rv is True:
-                            rv = None
-        elif any(a.is_infinite for a in Add.make_args(n)):
-            # (inf or nan)/x != 0
-            rv = False
-        if rv is not None:
-            return rv
+    from sympy.assumptions import Q
+    rel = Q.eq(lhs, rhs).canonical
+    return Q.eq.eval((rel.lhs, rel.rhs))
```

Comment:
This change seems to mess up the levels. The idea of `is_eq` was that it should be on a level below Relational. It was supposed to be a function that did not create new Basic instances and just answers a query.

If it is made redundant here then maybe that is fine but I don't (immediately) see any advantage in implementing all of the handler logic in `Q.eq` rather than in `is_eq`.